docs/how-tos/scaling.rst

```diff
@@ -0,0 +1,133 @@
+
+And all production deployments will need a `persistent Clickhouse cluster`_.
+
+Preparing the LMS workers
+=========================
+
+Before deploying Aspects, we recommend :ref:`event_bus`.
+
+Or, if you'd rather use celery, we recommend boosting the number of LMS workers to prepare the celery queue for the high-volume of Aspects tasks.
+
+These plugins can be used to configure and enable autoscaling for the LMS and CMS. See their READMEs for details:
+
+* `tutor-contrib-pod-autoscaling`_ : for single instance deployments
+* `tutor-contrib-grove`_ : for multiple instance deployments
+
+We also recommend configuring the LMS to use the high-priority celery queue for Aspects tasks (`platform-plugin-aspects event sinks`_, `event-routing-backends xAPI tasks`_). This leaves the low-priority queue clear for other LMS tasks.
+
+.. code-block:: yaml
+
+  TBD -- implement configurable queue for Aspects tasks
+
+Scaling Ralph
+=============
+
+Without scaling, Ralph can be a bottleneck in the Aspects data pipeline. If you do not need an LRS at all, you can disable Ralph and use :ref:`quick-start-vector` instead.
+
+But if you will ever need an LRS, it's better to start with Ralph enabled and autoscaled.
+
+Ralph runs CPU-intensive operations, so we recommend scaling mainly on CPU:
+
+.. code-block:: yaml
+
+  TBD -- implement RALPH autoscaling
+
+.. _scaling-clickhouse:
+
+Scaling Clickhouse
+==================
+
+We recommend using a Clickhouse service provider to manage your production cluster.
+
+Aspects `avoids using experimental Clickhouse features`_, and so is suitable for use with cloud providers. Cloud hosting also provides support, automated backups, and autoscaling. See :ref:`clickhouse-cluster` for details.
+
+However, if you decide to run your own Clickhouse instance, you will need to take into account:
+
+* horizontal and vertical scaling
+* replication and quorum
+* data storage requirements over time
+
+References:
+
+* `Clickhouse Operator`_: Helm charts, docs and examples from Altinity
+* `Clickhouse Keeper`_: recommended replication setup
+
+Small deployments can start with the following set up, and scale later:
+
+* 1 Clickhouse Keeper node, see `04-replication-zookeeper-01-minimal.yaml`_
+* 1 Clickhouse node, see `03-persistent-volume-01-default-volume.yaml`_
+
+For large deployments, we recommend:
+
+* 3 Clickhouse Keeper nodes to form the quorum: see `02-extended-3-nodes.yaml`_
+* N Clickhouse nodes to perform the replication.
+
+  If your k8s provider supports resizable volumes, see `03-persistent-volume-05-resizeable-volume-2.yaml`_
+  Otherwise, see `03-persistent-volume-02-pod-template.yaml`_
```

Comment:
Needs to be added to harmony, cf https://github.com/openedx/openedx-aspects/issues/80